src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -520,6 +532,13 @@ CHIP_ERROR AdvertiserMinMdns::GetCommissionableInstanceName(char * instanceName,
                                                      instanceName, maxLength);
 }
 
+CHIP_ERROR AdvertiserMinMdns::UpdateCommissionableInstanceName()
+{
+    uint64_t random_instance_name = chip::Crypto::GetRandU64();
+    memcpy(&mCommissionableInstanceName[0], &random_instance_name, sizeof(mCommissionableInstanceName));
```

Comment:
could we static_assert that sizeof(mCommissionableInstanceName) >= sizeof(random_instance_name) to avoid potentially copying stack data?

Reply:
Yes, absolutely.  In fact, we should static_assert equality.